### ISSUE TITLE
Created new app and shopify serve doesn't work

### ISSUE THREAD

**Issue by joshbedo** [Shopify/shopify-cli#1149]
After running `shopify create node` im trying to run `shopify serve` and I'm getting `✗ Couldn't verify your store`. I have no clue how to fix it and I can't find any information about it. I don't see anything in the development store saying it needs verified and the app API key/password match. errors should tell you more about whats going on and link to somewhere in documentation on how to fix it. it would be nice to have something like `<command> --verbose` so we can see the logging when the error happens.

**Comment by joshbedo** [comment]
Fixed: had to run `shopify connect` first and connect the app to the store for some reason it wasn't working when the app was created. It would be nice if the error said `Couldn't connect to store please run command 'shopify connect'`

**Comment by iamameme** [comment]
> I get `shopify connect was not found`.
> 
> Damn, do I hate when you run into errors just by following the quickstart guide.
> 
> Update: It looks like there was a major, breaking release a few days ago and no one bothered to update the docs or talk on how to use it somewhere.

Any update on this? I just started as well and am running into the same issue.

**Comment by S-YOU** [comment]
it is `shopify node connect` or `shopify rails connect`

**Comment by shredding** [comment]
Yields:

```
This command ran with ID: 59148
Please include this information in any issues/report along with relevant logs
/usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-ui/lib/cli/ui/prompt.rb:192:in `ask_interactive': insufficient options (ArgumentError)
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-ui/lib/cli/ui/prompt.rb:100:in `ask'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/tasks/ensure_env.rb:32:in `fetch_org'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/tasks/ensure_env.rb:17:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/task.rb:7:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/connect.rb:13:in `default_connect'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/project_types/node/commands/connect.rb:12:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/sub_command.rb:13:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/command.rb:19:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:16:in `block (2 levels) in call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:15:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/executor.rb:14:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/core/executor.rb:15:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/core/entry_point.rb:24:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/core/monorail.rb:20:in `log'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/lib/shopify-cli/core/entry_point.rb:23:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/bin/shopify:33:in `block in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	from /usr/local/Cellar/shopify-cli/2.0.1/gems/shopify-cli-2.0.1/bin/shopify:32:in `<top (required)>'
	from /usr/local/bin/shopify:6:in `load'
	from /usr/local/bin/shopify:6:in `<main>'
```

**Comment by aminDaryan** [comment]
Im still getting the error even after connection!

**Comment by aminDaryan** [comment]
I am still getting the  Couldn't verify your store even after connection. These are what are in my environments after connection
SHOPIFY_API_KEY, SHOPIFY_API_SECRET, SCOPES

**Comment by liambolling** [comment]
This is completely broken for me. Nobody can get started making apps at the moment and there is no communication from Shopify

**Comment by shredding** [comment]
For me it's now even more fundamentally broken, with `2.1.0` i can't even login anymore - the website gives me auth successful, but cli:

```
shopify login
This command ran with ID: 72483
Please include this information in any issues/report along with relevant logs
/usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:226:in `post_request': Invalid 'subject_token' value: invalid (ShopifyCli::IdentityAuth::Error)
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:208:in `post_token_request'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:202:in `request_exchange_token'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:181:in `block in request_exchange_tokens'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:180:in `each'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:180:in `request_exchange_tokens'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/identity_auth.rb:69:in `authenticate'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/commands/login.rb:33:in `call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/command.rb:25:in `call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:16:in `block (2 levels) in call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:15:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:14:in `call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/core/executor.rb:15:in `call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/core/entry_point.rb:24:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/core/monorail.rb:20:in `log'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/lib/shopify-cli/core/entry_point.rb:23:in `call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/bin/shopify:33:in `block in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	from /usr/local/Cellar/shopify-cli/2.1.0/gems/shopify-cli-2.1.0/bin/shopify:32:in `<top (required)>'
	from /usr/local/bin/shopify:6:in `load'
	from /usr/local/bin/shopify:6:in `<main>'
```

This feels really like an early alpha of a seed fund startup and not like the backbone of a stock listed platform.

**Comment by andyvu92** [comment]
I was able to solve the issue. Here is what I did:

1. Follow all the steps here https://shopify.dev/apps/getting-started/create up to 'shopify node serve'
2. Go to https://ngrok.com/, create an account and download ngrok terminal, then run it
3. After registering an account with ngrok, you should see instruction page. WIthin **Connect your account**, copy the command and paste into the ngrok terminal. Note that ./ngrok is more like path/to/ngrok that you just downloaded
4. Then within ngrok terminal, run **ngrok http 8081** (port 8081 is defined in the next app server.js created by shopify
5. Afterwards, you should see something like this
----------------------------------------------------------------------------
Session Status                online                                         Account                      ******** (Plan: Free)                                                                      Version                       2.3.40                                             Region                        *********                                                                        Web Interface                http://127.0.0.1:4040                                                    
Forwarding                    http://85cc899d9c8d.ngrok.io -> http://localhost:8081                                     
Forwarding                    https://85cc899d9c8d.ngrok.io -> http://localhost:8081
-------------------------------------------------------------------------------
6. Now go back to https://partners.shopify.com/ and navigate to Apps, select your new App and navigate to App setup
7. Within **URLs** section, fill in as follow (replace origin with your ngrok https address):
-------------------------------------------------------------------------------
App URL: https://85cc899d9c8d.ngrok.io/
Allowed redirection URL(s):  https://85cc899d9c8d.ngrok.io/
                                             https://85cc899d9c8d.ngrok.io/auth/callback
--------------------------------------------------------------------------------
8. Open a new terminal, navigate to your app folder and run 'npm run dev' to start next app server. You should see something like this:
--------------------------------------------------------------------------------
Event - compiled successfully
> Ready on http://localhost:8081
-------------------------------------------------------------------------------
9. Now go back to https://partners.shopify.com/ and navigate to Apps, select your new App. Find **Test your app** section and select store (if you don't have a store yet, create one for development). Click on the Store that you want to install the app and wait for it to process. When it's done, you should see something like this
![Screenshot (1)](https://user-images.githubusercontent.com/29296255/128196491-2edab846-020d-4772-8a98-befa3192a061.png)
10. Now try to change the text in pages/index.js in your local app folder then save. The live shopify app should be rerendered with the new changes. 
P/S: make sure that every time you work on the app, both ngrok server and next app need to be running

**Comment by wickywills** [comment]
So are the Shopify devs going to jump in here and address these issues? Still happening Sept 2021. The docs are a complete mess and very hard to follow. I too just tried to follow the basic setup guide and ran into problems - how does anyone develop for this platform!?

**Comment by dalynsw** [comment]
MacBook-Pro:shopify jay$ shopify node create
? App name
> my first app
? What type of app are you building? (You chose: Public: An app built for a wide merchant audience.)
This command ran with ID: 51049                                                                                                                                                                    
Please include this information in any issues/report along with relevant logs                                                                                                                      
/usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:225:in `post_request': Invalid 'subject_token' value: invalid (ShopifyCli::IdentityAuth::Error)
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:207:in `post_token_request'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:201:in `request_exchange_token'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:180:in `block in request_exchange_tokens'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:179:in `each'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:179:in `request_exchange_tokens'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/identity_auth.rb:68:in `authenticate'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/partners_api.rb:71:in `block in access_token'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/db.rb:87:in `get'
	from /usr/local/Cellar/ruby/3.0.2/lib/ruby/3.0.0/forwardable.rb:238:in `get'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/partners_api.rb:70:in `access_token'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/partners_api.rb:64:in `api_client'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/partners_api.rb:39:in `block in query'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/util.rb:172:in `retry_after'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/partners_api.rb:40:in `query'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/partners_api/organizations.rb:6:in `fetch_all'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/tasks/select_org_and_shop.rb:28:in `organizations'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/tasks/select_org_and_shop.rb:39:in `get_organization'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/tasks/select_org_and_shop.rb:11:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/task.rb:7:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/project_types/node/forms/create.rb:13:in `ask'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/form.rb:13:in `ask'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/project_types/node/commands/create.rb:22:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/sub_command.rb:13:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/command.rb:19:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:16:in `block (2 levels) in call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `block (2 levels) in with_logging'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:36:in `block in with_logging'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-ui/lib/cli/ui.rb:176:in `log_output_to'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:35:in `with_logging'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:15:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `block (2 levels) in with_traps'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:44:in `block in with_traps'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:55:in `twrap'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:43:in `with_traps'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/executor.rb:14:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/core/executor.rb:15:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/core/entry_point.rb:24:in `block in call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/core/monorail.rb:32:in `log'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/lib/shopify-cli/core/entry_point.rb:23:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/bin/shopify:33:in `block in <top (required)>'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	from /usr/local/Cellar/shopify-cli/2.4.0/gems/shopify-cli-2.4.0/bin/shopify:32:in `<top (required)>'
	from /usr/local/bin/shopify:6:in `load'
	from /usr/local/bin/shopify:6:in `<main>'

**Comment by anisometropie** [comment]
After a lot of trial and error, I found a solution.
`shopify node create` would always give me an error we I run `shopify node serve`

the solution is to run it with the store parameter
`shopify node create --STORE=xxx.myshopify.com`

**Comment by scottkatzelnick** [comment]
> So are the Shopify devs going to jump in here and address these issues? Still happening Sept 2021. The docs are a complete mess and very hard to follow. I too just tried to follow the basic setup guide and ran into problems - how does anyone develop for this platform!?

I'm currently building a custom app for my company's store and couldn't agree more. The docs look great, but if half the amount of thought went into developing cohesive, carefully considered up-to-date examples, thorough technical explanations with actual code laying out all options, and a much more straightforward categorization of all their different API's, as much as they put into styling them, I would have been able to finish this project months ago!

If all else fails, nuke your current project by copying all the necessary folders, files, and configuration to a temp directory. Make sure to show all hidden files so you don't miss anything. Also, create a login that is not already tied to a production Shopify site, as this can cause problems. Next, use the CLI to create a new project from scratch; first using `shopify app create node` then once the project finishes building, using `shopify login --store=<ENTER STORE URL>` and then try to `shopify app serve`. If this fails with the "could not verify your store" error, you also need to use the `shopify app connect` command to get it working. After all this, systematically bring back the files, folders, and configuration you have saved in the temp file.

It feels like the folks creating these docs aren't developers actively using any of it. How could so many users have so many more questions after reading docs than before? I find that skimming the docs to get the basic idea, then finding an actual fully-fledged example or repo, if possible, really connects the dots the way the documentation does not. Even if they added an example repo with each API, that would solve so many issues for people. They also need to show how to create it using express since that is so much more widely used and an up-to-date GraphQL client that carries more weight and has a better configuration like @apollo/client, not apollo-boost for a quick implementation. We're creating a custom Shopify app. Most of our apps are not portfolio projects, these are actual clients' solutions, and we need full explanations with complete examples. Just force your documentation folks to create, let's say, a fully functional subscription app using the docs as is, with all the required APIs (Subscription API, Selling Plan API, app extensions, etc.), and you'll see where the documentation is lacking. I'm only passionate cause it's a great platform  🙂

**Comment by xNikkox** [comment]
here we are in 2022 and these problems are still relevant what a mess.

**Comment by PierreTurnbull** [comment]
I had a similar problem. After doing `npm run dev` for the first time and installing my app on my partners shop, I stopped the server and relaunched it with the same command : `npm run dev`, but it didn't work anymore. I had some weird error like this (it was not always the same at each try): 
```
━━━━━━ Error ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    
    The Partners GraphQL API responded unsuccessfully with the HTTP status 504 and errors:
    
    {}
          

    Stack trace:    at                            graphql-request/src/index.ts:410      throw new ClientError(
    at step                       graphql-request/dist/index.js:63                            
    at next                       graphql-request/dist/index.js:44                            
    at fulfilled                  graphql-request/dist/index.js:35                            
    at processTicksAndRejection…  node:internal/process/task_queues:96                        

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```
I deleted the app in my partners account and did the process again with `npm run dev -- --reset`, installed the app again, stopped the server again and relaunched it, and this time the error did not show. I can't explain why it did not work at first or why it did work the second time.
